bin/embark

```diff
@@ -1,15 +1,1023 @@
+  PkgJsonLocal.prototype.log.call(this);
+  this.logMissingEmbarkDep();
+};
+
+PkgJsonLocalExpected.prototype.logMissingEmbarkDep = function () {
+  if (this.json && !this.embarkDep) {
+    reportMissingEmbarkDep(this.filepath, this.dirname, this.loglevel);
+    exitWithError();
+  }
+};
+
+PkgJsonLocalExpected.prototype.logMissingFile = function () {
+  // PkgJson.prototype NOT PkgJsonLocal.prototype
+  if (PkgJson.prototype.logMissingFile.call(this)) {
+    reportMissingFile_PkgJsonLocalExpected(this.dirname, this.loglevel);
+    exitWithError();
+  }
+};
+
+PkgJsonLocalExpected.prototype.setEmbarkDep = function () {
+  if (isObject(this.json)) {
+    if (this.json.dependencies) {
+      this.embarkDep = this.json.dependencies.embark;
+    } else if (this.json.devDependencies) {
+      this.embarkDep = this.json.devDependencies.embark;
+    }
+  }
+};
+
+PkgJsonLocalExpected.prototype.setup = function () {
+  PkgJsonLocal.prototype.setup.call(this);
+  this.setEmbarkDep();
+  return this;
+};
+
+// -- loggers ------------------------------------------------------------------
+
+var embarklog = npmlog;
+embarklog.heading = 'embark';
+
+var _logged = false;
+function logged(which) {
+  var embarklog_which = embarklog[which];
+  return function () {
+    _logged = true;
+    embarklog_which.apply(embarklog, arguments);
+  };
+}
+
+embarklog.error = logged('error');
+embarklog.info = logged('info');
+embarklog.warn = logged('warn');
+
+function blankLineMaybe(which) {
+  if (_logged) {
+    console[which]();
+  }
+}
+
+var isNpmRun = process.env.hasOwnProperty('npm_lifecycle_script');
+function blankLineTrailingMaybe(which) {
+  if (isNpmRun) {
+    console[which]();
+  }
+}
+
+// -- processors ---------------------------------------------------------------
+
+function checkPkg(pkgDir, scopes) {
+  var errors;
+  try {
+    var config = {packageDir: pkgDir};
+    if (scopes) {
+      config.scopeList = scopes;
+    }
+    var checked = checkDeps.sync(config);
+    if (checked.error.length) {
+      errors = checked.error;
+    }
+  } finally {
+    // eslint-disable-next-line no-unsafe-finally
+    return errors;
+  }
+}
+
+function parseJson(filepath) {
+  var parsed;
+  try {
+    parsed = require(filepath);
+  } finally {
+    // eslint-disable-next-line no-unsafe-finally
+    return parsed;
+  }
+}
+
+function parseRange(range) {
+  var parsed;
+  try {
+    parsed = semver.Range(range).range;
+  } finally {
+    // eslint-disable-next-line no-unsafe-finally
+    return parsed;
+  }
+}
+
+function realpath(filepath) {
+  var resolved;
+  try {
+    resolved = fs.realpathSync(filepath);
+  } finally {
+    // eslint-disable-next-line no-unsafe-finally
+    return resolved;
+  }
+}
+
+function thisEmbark() {
+  return (new EmbarkBin(__filename)).handle();
+}
+
+function whenNoShim() {
+  var noShim = false;
+  if (process.env.EMBARK_NO_SHIM) {
+    noShim = true;
+  }
+  var idx = process.argv.indexOf('--no-shim');
```

Comment:
I guess if we add a no-shim option, all commands should have `--no-shim` in the help.
I can see there is already the env var to specify no shim.
Why having two ways?

Reply:
I guess it is personal preference but I prefer to have 1 way only of doing things, that avoid confusion (at least for me) and keep the program simpler I believe.

Up to you

Reply:
Good point — I'll remove the cli option.

Reply:
Done.